### PR TITLE
chore(deps): update wish and keygen

### PR DESCRIPTION
**Pull request by bashbunni**
It's broken right now, you can see the tests failing. This is a new branch off of the `fix/dependabot` lint fixes. The `fix/dependabot` changes have passing tests, just this one with the updated `wish` and `keygen` versions... Would be open to pairing on it if your schedule allows :)

### CODE REVIEW

**Review by caarlos0 — APPROVED** [review]

**Review by maaslalani — APPROVED** [review]

File: cmd/import_keys.go [hunk 1]
```diff
@@ -217,15 +217,12 @@ func restoreFromReader(r io.Reader, dd string) error {
...
 		keypath+".pub",
 		ssh.MarshalAuthorizedKey(signer.PublicKey()),
 		0o600,
-	); err != nil {
-		return err
-	}
-
-	return nil
+	)
+	return err
```
[the remark this answers is not in the document]
Comment (bashbunni) [reply cmd/import_keys.go:225]: Oh so true! Thanks

File: client/client.go [hunk 2]
```diff
@@ -81,7 +81,8 @@ func NewClient(cfg *Config) (*Client, error) {
 			if err != nil {
 				return nil, err
 			}
-			_, err = keygen.NewWithWrite(filepath.Join(dp, "charm"), []byte(""), cfg.KeygenType())
+
+			_, err = keygen.New(filepath.Join(dp, "charm"), keygen.WithKeyType(cfg.KeygenType()), keygen.WithWrite())
```
Comment (aymanbagabas) [inline client/client.go:85]:
```suggestion
			_, err = keygen.New(filepath.Join(dp, "charm_"+cfg.KeygenType().String()), keygen.WithKeyType(cfg.KeygenType()), keygen.WithWrite())
```

Old versions of keygen append the key type to path

File: cmd/post_news.go [hunk 3]
```diff
@@ -26,11 +26,11 @@ var (
 				cfg.DataDir = serverDataDir
 			}
 			sp := filepath.Join(cfg.DataDir, ".ssh")
-			kp, err := keygen.NewWithWrite(filepath.Join(sp, "charm_server"), []byte(""), keygen.Ed25519)
+			kp, err := keygen.New(filepath.Join(sp, "charm_server"), keygen.WithKeyType(keygen.Ed25519), keygen.WithWrite())
```
Comment (aymanbagabas) [inline cmd/post_news.go:29]:
```suggestion
			kp, err := keygen.New(filepath.Join(sp, "charm_server_ed25519"), keygen.WithKeyType(keygen.Ed25519), keygen.WithWrite())
```

File: cmd/serve.go [hunk 4]
```diff
@@ -48,11 +48,11 @@ var (
 				cfg.DataDir = serverDataDir
 			}
 			sp := filepath.Join(cfg.DataDir, ".ssh")
-			kp, err := keygen.NewWithWrite(filepath.Join(sp, "charm_server"), []byte(""), keygen.Ed25519)
+			kp, err := keygen.New(filepath.Join(sp, "charm_server"), keygen.WithKeyType(keygen.Ed25519), keygen.WithWrite())
```
Comment (aymanbagabas) [inline cmd/serve.go:51]:
```suggestion
			kp, err := keygen.New(filepath.Join(sp, "charm_server_ed25519"), keygen.WithKeyType(keygen.Ed25519), keygen.WithWrite())
```

File: testserver/testserver.go [hunk 5]
```diff
@@ -34,12 +34,13 @@ func SetupTestServer(tb testing.TB) *client.Client {
 	cfg.HTTPPort = randomPort(tb)
 	cfg.HealthPort = randomPort(tb)
 
-	kp, err := keygen.NewWithWrite(filepath.Join(sp, "charm_server"), []byte(""), keygen.Ed25519)
+	kp, err := keygen.New(filepath.Join(sp, "charm_server"), keygen.WithKeyType(keygen.Ed25519), keygen.WithWrite())
```
Comment (aymanbagabas) [inline testserver/testserver.go:37]:
```suggestion
	kp, err := keygen.New(filepath.Join(sp, "charm_server_ed25519"), keygen.WithKeyType(keygen.Ed25519), keygen.WithWrite())
```

**Review by aymanbagabas — APPROVED** [review]

**Pull request merged**